xarray/core/alignment.py

```diff
@@ -99,6 +105,21 @@ def align(
         obj, = objects
         return (obj.copy(deep=copy),)
 
+    if join == "override":
```

Comment:
Could the logic from this block be moved to a helper function? `align` is already getting pretty long.